### PR TITLE
UX/reliability changes.

### PR DESCRIPTION
**Pull request by concretevitamin**
Many changes from a few weeks ago.

UX
- Make entrypoint optional -- no more `sky launch <flags> ''`, simply do `sky launch <flags>`
- Several reworded logging messages
- After a `spot launch`, no longer print the first few entries from `spot status` as it's too verbose

Reliability
- Handle rate limit exceeded errors from GCP (triggered when launching a bunch of things)
- Fail the runtime setup if ray patching somehow failed

### CODE REVIEW

File: sky/execution.py [hunk 1]
```diff
@@ -228,15 +227,9 @@ def _execute(
         # UX: print live clusters to make users aware (to save costs).
         # Needed because this finally doesn't always get executed on errors.
         # Disable the usage collection for this status command.
```
Comment (Michaelvll) [inline sky/execution.py:229]: nit: Move these comments after L230?

File: sky/execution.py [hunk 2]
```diff
@@ -228,15 +227,9 @@ def _execute(
...
-        env = dict(os.environ,
-                   **{env_options.Options.DISABLE_LOGGING.value: '1'})
-        if cluster_name == spot.SPOT_CONTROLLER_NAME:
-            # For spot controller task, it requires a while to have the
-            # managed spot status shown in the status table.
-            time.sleep(0.5)
-            subprocess_utils.run(
-                f'sky spot status | head -n {_MAX_SPOT_JOB_LENGTH}')
-        else:
+        if cluster_name != spot.SPOT_CONTROLLER_NAME:
```
Comment (Michaelvll) [inline sky/execution.py:230]: Should we add a comment here about why we do not show `status` or `spot status` for the controller?

File: sky/templates/spot-controller.yaml.j2 [hunk 3]
```diff
@@ -33,6 +33,10 @@ setup: |
     source ~/google-cloud-sdk/path.bash.inc > /dev/null 2>&1 && \
     echo 'source ~/google-cloud-sdk/path.bash.inc > /dev/null 2>&1' >> ~/.bashrc)
 
+  {% if is_dev %}
+  echo 'export SKYPILOT_DEV=1' >> ~/.bashrc
+  {% endif %}
```
Comment (Michaelvll) [inline sky/templates/spot-controller.yaml.j2:38]: Just curious, why is this needed? Our `envs` section below should cover all the `setup` and `run` commands on the controller. Is it for logging into the controller through ssh and debugging?

**Review by Michaelvll — APPROVED** [review]
Nice! Thank you for the fixes @concretevitamin !

File: sky/templates/spot-controller.yaml.j2 (hunk 3 above)
Comment (concretevitamin) [reply sky/templates/spot-controller.yaml.j2:39]: Exactly for that use case; added a comment.

File: sky/execution.py (hunk 2 above)
Comment (concretevitamin) [reply sky/execution.py:227]: Done.

File: sky/execution.py (hunk 1 above)
Comment (concretevitamin) [reply sky/execution.py]: Done.

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll - PTAL.

**Review by Michaelvll — APPROVED** [review]
Thank you for the quick fix! LGTM.
